tests/app/questionnaire/test_router.py

```diff
@@ -510,6 +547,31 @@ def test_return_to_final_summary_questionnaire_is_complete(self):
 
         self.assertEqual(url_for("questionnaire.submit_questionnaire"), next_location)
 
+    def test_return_to_final_summary_section_is_in_progress(self):
+        self.schema = load_schema_from_name("test_submit_with_summary")
+        self.progress_store = ProgressStore(
+            [
+                {
+                    "section_id": "default-section",
+                    "list_item_id": None,
+                    "status": CompletionStatus.IN_PROGRESS,
+                    "block_ids": ["radio", "dessert", "dessert-confirmation"],
+                }
+            ]
+        )
+        current_location = Location(
+            section_id="default-section", block_id="dessert-confirmation"
+        )
+        routing_path = RoutingPath(
+            ["radio", "dessert", "dessert-confirmation", "numbers"],
+            section_id="default-section",
+        )
+        next_location = self.router.get_next_location_url(
+            current_location, routing_path, return_to="final-summary"
+        )
+
+        self.assertIn("/questionnaire/numbers/?return_to=final-summary", next_location)
```

Comment:
I'm a bit torn on these, although these test are correct and perfectly valid, I don't think it is enough to validate that the query string is persisted. These tests are essentially unit testing that, if you provide a `return_to` to `get_next_location_url`, it behaves as expected rather than it is persisted between requests. 

If the block handler was not passed the `return_to`, these unit tests would still pass, but the URL wouldn't have the query string at run time. Therefore, we should have an integration/functional test to assert that it is indeed persisted between requests. I couldn't see any integration/function tests for it.

Reply:
I had a similar thought pattern, but came down on the side of not having them, as it was harnessing something that we already had (i.e the return_to ). However it was only by a nano smidgen so will stick some extra tests in